src/mm.ts

```diff
@@ -805,8 +808,10 @@ function makeMarketUpdateInstructions(
       mangoAccount.getOpenOrdersKeysInBasket(),
```

Comment:
It would be better to use just the open orders in the basket and not also add in zero keys for the empty ones. PlacePerpOrder2 supports that

Reply:
I added this helper code in client. You have to `yarn upgrade @blockworks-foundation/mango-client` and it should be available: https://github.com/blockworks-foundation/mango-client-v3/blob/e77e957a7f00041561ac2670170c428407bb90ed/src/MangoAccount.ts#L1077

Reply:
Thanks @dafyddd, have pushed a change that does this.